x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { intersection } from 'lodash';
+import { leftJoin } from '../../../common/utils/left_join';
+import { rangeFilter } from '../helpers/range_filter';
+import { Job as AnomalyDetectionJob } from '../../../../ml/server';
+import { PromiseReturnType } from '../../../typings/common';
+import { IEnvOptions } from './get_service_map';
+import { APM_ML_JOB_GROUP_NAME } from '../../../common/ml_job_constants';
+
+type ApmMlJobCategory = NonNullable<ReturnType<typeof getApmMlJobCategory>>;
+const getApmMlJobCategory = (
+  mlJob: AnomalyDetectionJob,
+  serviceNames: string[]
+) => {
```

Comment:
Weren't we aiming to use the parameterised jobs for 7.9 in which case this is no longer needed?

Reply:
it's out of scope for the 7.8 fix, so if we go down that route for 7.9, we'll remove this

Reply:
Oh, sorry. Missed this was for 7.8!